client/app/private-database/database/dump/private-database-database-dump.controller.js

```diff
@@ -126,17 +107,17 @@ angular
             this.database.waitRestore = true;
             this.dumpIdRestoring = opts.dumpId;
 
-            this.findItemIndex(opts.dumpId)
-                .then((idx) => {
-                    this.$scope.dumpsDetails[idx].waitRestore = true;
+            this.dumps.filter((dump) => dump.id === opts.id)
+                .forEach((dump) => {
```

Comment:
.map would be better there no ?

Reply:
this.dumps is not newly assign, it would have the same behaviour, but this is a bit more readable